app/assets/javascripts/components/common/wiki_select.jsx

```diff
@@ -104,15 +104,30 @@ const WikiSelect = createReactClass({
         this.props.onChange(this.props.multi ? [] : {});
       }
     };
+
     const options = this.props.options;
-    const filterOptions = function (val) {
+    // Two types of matching for optimizing results.
+    // If the input is less than four characters, it will be matched from the beginning of the string.
+
+    const filterOptionsLong = function (val) {
       return options.filter(wiki =>
         wiki.label.toLowerCase().includes(val.toLowerCase())
       ).slice(0, 10); // limit the options for better performance
     };
 
+  // If the input is less than four characters, it will be matched anywhere in the string
+
+    const filterOptionsShort = function (val) {
+      return options.filter(wiki =>
+        wiki.label.toLowerCase().startsWith(val.toLowerCase())
+      ).slice(0, 10); // limit the options for better performance
+    };
+
     const loadOptions = function (inputValue, callback) {
-      callback(filterOptions(inputValue));
+      if (inputValue.length < 4) {
```

Comment:
I like this strategy! Since the project domains all start with "wiki", this should accurately filter to an appropriate set of wikis in almost all cases. The one thing I could think of is that, for wikidata (and three other unique projects), we put them at the top of the list currently, so that when you type `w`you see `www.wikidata`, `www.wikisource`, `commons`, and `incubator` at the top. However, `w` is a valid language code, so we don't want to special-case that.

But since the problem we're solving for is language-code collisions where a two letter code is part of a three letter code (but not the first part), I think we could safely move the "short" cutoff to `< 3`.

Reply:
:D